llama_index/indices/document_summary/retrievers.py

```diff
@@ -176,7 +179,12 @@ def _get_embeddings(
         self._index._service_context.callback_manager.on_event_end(
             CBEventType.EMBEDDING,
             payload={
-                EventPayload.CHUNKS: [x for x in nodes if node.embedding is not None]
+                EventPayload.CHUNKS: [
+                    node.get_content(metadata_mode=MetadataMode.EMBED)
+                    for node in nodes
+                    if node.embedding is not None
+                ],
+                EventPayload.EMBEDDINGS: result_embeddings,
```

Comment:
What's the reason we are only returning nodes with a non-null embedding?

I saw this hook is returning nodes rather than text, so I changed to be consistent with the other embeddings.

Reply:
Feels unexpected to have the embedding hook on a particular retriever. What is the reason for this as opposed to having it only on `llama_index.embeddings.base.BaseEmbedding`?

Reply:
we definitely can rely on the base embedding callback, this one seems redundant. We can probably even remove this.

Reply:
Cool, mind if I remove as part of this PR? @logan-markewich

Reply:
Still not clear on why we are only returning nodes with a non-null embeddings.

Reply:
Yea go for it! Tbh I have no idea either right now lol